.github/workflows/python-publish.yml

```diff
@@ -22,6 +22,18 @@ jobs:
       run: |
         python -m pip install --upgrade pip
         pip install setuptools wheel twine
+    # Here we don't use ${GITHUB_REF_NAME#v} since it won't parse in the name field.
+    - name: Bump package version to ${{ github.ref_name }}
+      run: |
+        git config user.name "GitHub Actions Bot"
+        git config user.email "<>"
+        pip install bump2version==1.0.1
+        # Here we don't use ${{ github.ref_name }} since github action variables can't do sed/cut/slicing
+        # Remove leading 'v' from the tag name
+        bump2version --new-version ${GITHUB_REF_NAME#v} minor
+        # must supply either patch minor or major see https://github.com/c4urself/bump2version/issues/244
+        # but it's an ignored argument argument
+        git show
```

Comment:
We also need a step to perform a git push at the very end (after pypi) upload is successful but being in a detached head mode in the CI is proving an issue. Going to park this for now :(